src/classes/BDI_DataImportService.cls

```diff
@@ -1725,66 +1725,81 @@ global with sharing class BDI_DataImportService {
     */
     private void importCampaigns() {
 
-        // since we don't currently fill in a campaign lookup or status on the DI record,
-        // there is no work to do in Dry Run.
-        if (isDryRun) {
-            return;
-        }
-
         // need to find all the Campaigns that we might need
-        Set<String> setStrCampaign = new Set<String>();
+        Set<String> uniqueCampaignNames = new Set<String>();
+        Set<Id> setCampaignId = new Set<Id>();
+
         for (DataImport__c dataImport : listDI) {
             // skip di's that already have an error
             if (dataImport.Status__c == BDI_DataImport_API.bdiFailed) {
                 continue;
             }
 
             if (dataImport.Donation_Campaign_Name__c != null) {
-                setStrCampaign.add(dataImport.Donation_Campaign_Name__c);
+                uniqueCampaignNames.add(dataImport.Donation_Campaign_Name__c);
             }
 
         }
 
         // add the existing campaigns to our map
-        for (Campaign cmp : [SELECT Id, Name FROM Campaign WHERE Name IN :setStrCampaign]) {
-            mapStrNameToCampaignId.put(cmp.Name, cmp.Id);
+        for (Campaign cmp : [SELECT Id, Name FROM Campaign WHERE Name IN :uniqueCampaignNames]) {
```

Comment:
I know you didn't write the logic here, but I feel like we should at least add `ORDER BY CreatedDate DESC` to make the selection predictable (though flawed). I'd also consider moving the query out of the for loop for clarity.